luavm/src/lib/lua_values/lua_closure.rs

```diff
@@ -1,20 +1,54 @@
+use crate::{stdlib::StdFunction, Vm};
 use gc::GcCell;
+use luacompiler::bytecode::Function;
 
 /// Represents a closure in Lua.
 #[derive(Trace, Finalize)]
 pub struct LuaClosure {
     index: usize,
+    reg_count: usize,
+    param_count: usize,
     args_count: GcCell<usize>,
     args_start: GcCell<usize>,
+    #[unsafe_ignore_trace]
+    handler: Option<fn(&mut Vm)>,
```

Comment:
What is the "handler"?

Reply:
We almost certainly don't want to do things in this way: it means that every struct is carrying around an extra word (the pointer) *plus* we have additional checks.

Generally (in Python-ish syntax) I do something like:

```python
class Obj: pass

class Func(Obj): pass

class UserFunc(Func): pass

class BuiltinFunc(Func): pass
```

In practise, I generally drop the intermediate class (`Func`) because it doesn't do much, but it's there conceptually.